ddtrace/tracer/writer.go

```diff
@@ -183,15 +184,15 @@ func (h *logTraceWriter) encodeSpan(s *span) {
 	first := true
 	for k, v := range s.Meta {
 		if first {
-			h.buf.WriteByte('"')
 			first = false
 		} else {
-			h.buf.WriteString(`,"`)
+			h.buf.WriteString(`,`)
 		}
-		h.buf.WriteString(k)
-		h.buf.WriteString(`":"`)
-		h.buf.WriteString(v)
-		h.buf.WriteString(`"`)
+		m, _ := json.Marshal(k)
```

Comment:
Let's find a better alternative. The whole point of this tedious function was to avoid `json.Marshal` for speed. Otherwise the whole thing could've been replaced by `json.Marshal` 🙂 

If we just want to escape some characters, I'm sure there's better ways in the standard library? Look deeper at the implementation of `json.Marshal` to find out. Here's a hint from its documentation:

```go
// String values encode as JSON strings coerced to valid UTF-8,
// replacing invalid bytes with the Unicode replacement rune.
// So that the JSON will be safe to embed inside HTML <script> tags,
// the string is encoded using HTMLEscape,
// which replaces "<", ">", "&", U+2028, and U+2029 are escaped
// to "\u003c","\u003e", "\u0026", "\u2028", and "\u2029".
// This replacement can be disabled when using an Encoder,
// by calling SetEscapeHTML(false).
```

Reply:
From what I've been told, the increase in the speed due to that json.Marshal call - is nothing and is acceptable

Reply:
I think you mean the decrease. Since the point here was to get rid of `json.Marshal`, adding it back here is weird. But the string encoding bit of `json.Marshal` is quite tricky to reproduce so let's leave it.

Reply:
Resolved, then.